pp/go/relabeler/head/head.go

```diff
@@ -750,16 +767,18 @@ func (h *Head) Concurrency() int64 {
 	return calculateHeadConcurrency(h.numberOfShards)
 }
 
-// readOnlyForEachShard run generic task on read only head without queue.
+// readOnlyForEachShard run generic task on read only head without queue on all shards.
 func (h *Head) readOnlyForEachShard(t *relabeler.GenericTask) {
-	for shardID := uint16(0); shardID < h.numberOfShards; shardID++ {
-		s := h.shards[shardID]
-		go func(sd *shard) {
-			t.ExecuteOnShard(sd)
-		}(s)
+	for _, s := range h.shards {
+		h.readOnlyOnShard(t, s)
 	}
 }
 
+// readOnlyOnShard run generic task on read only head without queue on specific shard.
+func (h *Head) readOnlyOnShard(t *relabeler.GenericTask, s *shard) {
+	go func(sd *shard) { t.ExecuteOnShard(sd) }(s)
```

Comment:
starting goroutine is a heavy operation. In this place better run task in caller goroutine